src/content/en/updates/2017/06/budget-api.md

```diff
@@ -0,0 +1,189 @@
+
+{# wf_updated_on: 2017-06-07 #}
+{# wf_published_on: 2017-06-07 #}
+{# wf_tags: javascript,push,budget,api #}
+{# wf_featured_image: /web/updates/images/generic/info.png #}
+
+# Introduction to the Budget API {: .page-title }
+
+The Push Messaging API enables us to send notifications to a user even when the
+browser is closed. Many developers want to be able to use this messaging to
+update and synchornise content without the browser being open, but the API has
+one important restriction: you must always display a notification for every
+single push message recieved.
+
+Being able to send a push message to synchronize data on a user's device can be
+extremely useful for users and developers, but allowing a web app to do work in
+the background without the user knowing is open to abuse.
+
+The [Budget
+](https://wicg.github.io/budget-api/)[API](https://wicg.github.io/budget-api/),
+is a new API designed to allow developers to perform background work without
+notifying the user, such as silent push or performing a background fetch. In
+Chrome version 60 and above you'll be able to start using this API and the
+Chrome team is eager to get feedback from developers.
+
+To allow developers to consume a user's resources in the background, the web
+platform is introducing the concept of a budget using the new Budget API. Each
+site has a set amount of resource that they can consume for background actions,
+such as a silent push, will depleats the budget. When the budget is spent,
+background actions can no longer be performed without user visibility or
+consent. The user agent will be responsible for determining budget assigned to a
+web app based on it's heuristics, for example budget allowance could be linked
+to user engagement.
+
+**TL;DR **The budget API allows to you to reserve budget, use budget, get a list
+of remaining budget and understand the cost of background operations
+
+## Reserving Budget
+
+In Chrome version 60 and above, the `navigator.budget.reserve()` method will be available
+without any flags.
+
+The `reserve` method allows you to request budget for a specific operation and
+it'll return a boolean to indicate if the budget could be reserved or not. If
+the budget was reserved, there is no need to notify the user of your background
+work.
+
+In the example of push notifications, you can attempt to reserve budget for a
+"silent-push" operation and if `reserve()` resolves with true, the operation is allowed,
+otherwise it'll return false and you'll need to show a notification
+
+```javascript
+self.addEventListener('push', event => {
+ const promiseChain = navigator.budget.reserve('silent-push')
+   .then((reserved) => {
+     if (reserved) {
+       // No need to show a notification.
+       return;
+     }
+
+     // Not enough budget is available, must show a notification.
+     return registration.showNotification(...);
```

Comment:
Let me see if I'm understanding this correctly? If there's no budget, I _must_ have a user interaction. Since you haven't taken fleshed out the notification in the example, I take that to mean that a basic notification will do and that I _don't_ need `{ actions: {} }`. Is this correct? 

You might want to say something about that either way.